src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4641,6 +4659,25 @@ private void ParseParameterModifiers(SyntaxListBuilder modifiers, bool isFunctio
 
                 modifiers.Add(modifier);
             }
+
+            bool shouldTreatAsModifier()
+            {
+                var resetPoint = this.GetResetPoint();
+
+                Debug.Assert(this.CurrentToken.Kind == SyntaxKind.IdentifierToken);
+                this.EatToken();
+
+                bool validAsModifier = IsParameterModifier(this.CurrentToken, isFunctionPointerParameter) ||
```

Comment:
It seems like `IsParameterModifier` always returns true when we have `Kind == Identifier` and `ContextualKind == Scoped`.

Reply:
I'm trying to understand whether the `ScanType && ...` condition which follows is dead code.